tests/runtime/test_oqc_provider.py

```diff
@@ -0,0 +1,162 @@
+# Copyright (C) 2024 qBraid
+#
+# This file is part of the qBraid-SDK
+#
+# The qBraid-SDK is free software released under the GNU General Public License v3
+# or later. You can redistribute and/or modify it under the terms of the GPL v3.
+# See the LICENSE file in the project root or <https://www.gnu.org/licenses/gpl-3.0.html>.
+#
+# THERE IS NO WARRANTY for the qBraid-SDK, as per Section 15 of the GPL v3.
+
+"""
+Unit tests for OQCProvider class
+
+"""
+from unittest.mock import Mock, patch
+
+import numpy as np
+import pytest
+
+from qbraid.programs import NATIVE_REGISTRY, ProgramSpec
+from qbraid.runtime import DeviceType, TargetProfile
+from qbraid.runtime.oqc import OQCDevice, OQCJob, OQCProvider
+from qbraid.transpiler import ConversionScheme
+
+try:
+    from qcaas_client.client import OQCClient, QPUTask
+
+    oqc_not_installed = False
+except ImportError:
+    oqc_not_installed = True
+
+pytest.mark.skipif(oqc_not_installed, reason="qcaas_client not installed")
+
+
+def braket_circuit():
+    """Returns low-depth, one-qubit Braket circuit to be used for testing."""
+    import braket.circuits  # pylint: disable=import-outside-toplevel
+
+    circuit = braket.circuits.Circuit()
+    circuit.h(0)
+    circuit.ry(0, np.pi / 2)
+    return circuit
+
+
+def cirq_circuit(meas=True):
+    """Returns Low-depth, one-qubit Cirq circuit to be used for testing.
+    If ``meas=True``, applies measurement operation to end of circuit."""
+    import cirq  # pylint: disable=import-outside-toplevel
+
+    q0 = cirq.GridQubit(0, 0)
+
+    def basic_circuit():
+        yield cirq.H(q0)
+        yield cirq.Ry(rads=np.pi / 2)(q0)
+        if meas:
+            yield cirq.measure(q0, key="q0")
+
+    circuit = cirq.Circuit()
+    circuit.append(basic_circuit())
+    return circuit
+
+
+def qiskit_circuit(meas=True):
+    """Returns Low-depth, one-qubit Qiskit circuit to be used for testing.
+    If ``meas=True``, applies measurement operation to end of circuit."""
+    import qiskit  # pylint: disable=import-outside-toplevel
+
+    circuit = qiskit.QuantumCircuit(1, 1) if meas else qiskit.QuantumCircuit(1)
+    circuit.h(0)
+    circuit.ry(np.pi / 2, 0)
+    if meas:
+        circuit.measure(0, 0)
+    return circuit
+
+
+def test_circuits():
+    """Returns list of test circuits for each available native provider."""
+    circuits = []
+    if "cirq" in NATIVE_REGISTRY:
+        circuits.append(cirq_circuit(meas=False))
+    if "qiskit" in NATIVE_REGISTRY:
+        circuits.append(qiskit_circuit(meas=False))
+    if "braket" in NATIVE_REGISTRY:
+        circuits.append(braket_circuit())
+    return circuits
+
+
+device_id = "qpu:uk:2:d865b5a184"
```

Comment:
Constants declared at top of files should generally use all caps e.g.

```python
DEVICE_ID = "qpu:uk:2:d865b5a184"
```

or can make it a fixture if used in every test function and and then pass directly as arg to test function e.g.

```python
@pytest.fixture
def device_id():
    yield ""qpu:uk:2:d865b5a184""

def test_oqc_provider_device(device_id):
   ...
```